src/openrct2/peep/Peep.cpp

```diff
@@ -1785,6 +1787,22 @@ Peep* Peep::Generate(const CoordsXYZ& coords)
 
     increment_guests_heading_for_park();
 
+#ifdef ENABLE_SCRIPTING
+    auto& hookEngine = OpenRCT2::GetContext()->GetScriptEngine().GetHookEngine();
+    if (hookEngine.HasSubscriptions(OpenRCT2::Scripting::HOOK_TYPE::GUEST_GENERATION))
+    {
+        auto ctx = OpenRCT2::GetContext()->GetScriptEngine().GetContext();
+
+        // Create event args object
+        auto obj = OpenRCT2::Scripting::DukObject(ctx);
+        obj.Set("id", peep->sprite_index);
```

Comment:
Should we add a `handled` bool so that plugins can set this to true if they do not want the guest to spawn?

Reply:
The time it calls hook is a bit late here but I agree that the hook should control the spawn.

Reply:
I don't think we should allow control at this point. A hook earlier when deciding to generate would be better for that. Then they can up the spawn rate or adjust it to suit. I don't want to move the hook to the start of the function to prevent consumers changes being overwritten.

Reply:
So what exactly is the point of this hook then? You could in theory just subscribe to tick and remember the last guest count if you wanna know how many new guests entered. Perhaps name some example on how this hook could be used?

Reply:
It gives you the entity id of the new guest. You can then do with that as you wish. There is currently a plug-in that iterates all entities every tick looking for a new entity so that it can assign a name based on twitch chat users. With this they just use the hook to assign the name.

Reply:
If there is another way to control guest generation then that's fine.

Reply:
You can already use the cheat game action to force spawn new guests. We don't at present expose anything else. ZehMatt suggested exposing a model that plugins can modify. That would improve performance over a hook as deciding to spawn a guest would be a function that is called relatively often compared to this hook where generation has already been decided.